### PR TITLE
[FRONT-108] Wallet only auth

### PR DESCRIPTION
**Pull request by juhah**
Sorry for the fairly sizeable PR but there are a lot of removed files due getting rid of much of the old login code. Also the change password option is not available anymore, so more code can go into the archives.

Now we have only login available with Metamask and WalletConnect:
Screenshot 2020-11-26 at 15 26 22

Some more details in comments 👇 To test, try logging in with Metamask, refusing access, refusing the challenge, etc. With WalletConnect, you can use https://test.walletconnect.org/ on your phone to scan the code (or in browser by pasting the code).

Screenshot 2020-11-26 at 15 25 17

Take a look also at the changed menus when signed in / signed out:
Screenshot 2020-11-26 at 15 34 39
Screenshot 2020-11-26 at 15 34 45

### CODE REVIEW

File: app/src/app/index.jsx [hunk 1]
```diff
@@ -59,21 +54,18 @@ const DashboardEditorAuth = userIsAuthenticated(DashboardEditor)
...
-    <Route path={routes.auth.signUp()} component={SignupPage} key="SignupPage" />,
-    <Route path={routes.auth.forgotPassword()} component={ForgotPasswordPage} key="ForgotPasswordPage" />,
-    <Route path={routes.auth.resetPassword()} component={ResetPasswordPage} key="ResetPasswordPage" />,
-    <Route exact path={routes.auth.register()} component={RegisterPage} key="RegisterPage" />,
+    // Redirect old paths to login
     <Redirect from="/login/auth" to={routes.auth.login()} key="LoginRedirect" />,
-    <Route exact path="/register/register" key="RegisterRedirect" render={forwardTo(routes.auth.register)} />,
-    <Route exact path="/register/resetPassword" key="ResetPasswordRedirect" render={forwardTo(routes.auth.resetPassword)} />,
-    <Redirect from="/register/forgotPassword" to={routes.auth.forgotPassword()} key="ForgotPasswordRedirect" />,
+    <Redirect from="/signup" to={routes.auth.login()} key="SignUpRedirect" />,
```
Comment (juhah) [inline app/src/app/index.jsx:62]: Left the old routes available but redirecting to login

File: app/src/auth/hooks/useWalletConnect.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                    connected = true
+                }
+
+                if (error) {
+                    reject(error)
+                }
+
+                if (!challengePending) {
+                    // Get provided accounts and chainId
+                    const { accounts /* , chainId */ } = payload.params[0]
```
Comment (juhah) [inline app/src/auth/hooks/useWalletConnect.js:79]: `chainId` could be used to verify network but it doesn't seem to matter when logging in

File: app/src/auth/hooks/useWalletConnect.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                })
+            }
+
+            if (!connector) {
+                throw new Error('connector hasn\'t been created yet')
+            }
+
+            // Subscribe to connection events
+            connector.on('connect', onConnect)
+            connector.on('disconnect', onDisconnect)
```
Comment (juhah) [inline app/src/auth/hooks/useWalletConnect.js:123]: There is no apparent way to unsubscribe from these events 🤷

**Comment by juhah** [comment]
Added display of login method in tablet / mobile:
Screenshot 2020-11-27 at 14 57 34

Screenshot 2020-11-27 at 14 57 43

This should change when opening menu: https://linear.app/streamr/issue/FRONT-167/streamr-layout-mobile-menu-fixes

**Comment by juhah** [comment]
FYI, identicon default avatars are not added here: https://linear.app/streamr/issue/FRONT-178/use-identicons-for-default-avatar-image

File: app/src/auth/components/AuthLayout/authLayout.pcss [hunk 4]
```diff
@@ -1,73 +1,5 @@
...
-  .footer {
-    flex-grow: 1;
-    margin-bottom: 4em;
-  }
-
-  .content {
-    flex-grow: 1;
-    justify-content: initial;
-  }
+  background: #E5E5E5;
```
Comment (tumppi) [inline app/src/auth/components/AuthLayout/authLayout.pcss:3]: This should be #F8F8F8. (I have done the same mistake when looking this from Figma. It quite often shows the default canvas background and not the selected page background color)

File: app/src/auth/i18n/en.po [hunk 5]
```diff
@@ -8,86 +8,36 @@ msgstr ""
...
-msgid "auth.resetPassword"
-msgstr "Reset password"
+msgid "auth.tryAgain"
+msgstr "Try again"
 
-msgid "auth.signUp.success.title"
-msgstr "Thanks for signing up!"
+msgid "auth.terms"
+msgstr ""
+"By connecting your wallet and using Streamr<br />"
```
Comment (tumppi) [inline app/src/auth/i18n/en.po:36]: Why `<br />` here? The text is on one line. A space should do here.

**Review by tumppi — APPROVED** [review]
Works like a charm! Could not break it. Well done 🎉!

File: app/src/auth/i18n/en.po (hunk 5 above)
Comment (juhah) [reply app/src/auth/i18n/en.po:36]: It's my lazy way of adding a line break in mobile, it is hidden here for tablet and up: 
https://github.com/streamr-dev/streamr-platform/pull/1135/files#diff-219f11e3e0df3742e50a30db61fc6067a89876206041626ecb697784197db3bfR82-R88

Comment (tumppi) [reply app/src/auth/i18n/en.po:36]: Oh. That makes sense. We should still add a space to show it on desktop 😄

Comment (juhah) [reply app/src/auth/i18n/en.po]: Ah, true - thanks, didn't see that 👍

**Review by mondoreale — APPROVED** [review]
Played around with it. It works pretty well, sir! 👍

Left a bunch of comments. Give them a thought.

Cheers!

File: app/src/auth/hooks/useWalletConnect.js (hunk 3 above)
Comment (juhah) [reply app/src/auth/hooks/useWalletConnect.js:124]: Yep, tried `off`, `removeListener` and `removeAllListeners` but they don't exist. I believe the point of WalletConnect is to persist the session for the duration (it keeps polling the account). Here we are killing the session after getting the signature, I suppose it's not designed for one-off usage.

File: app/src/auth/hooks/useMetamask.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+        return token
+    }, [])
+
+    return useMemo(() => ({
+        connect,
+    }), [
+        connect,
+    ])
+}
```
[the remark this answers is not in the document]
Comment (juhah) [reply app/src/auth/hooks/useMetamask.js:26]: Yea, can change it to simplify... I had another variable returned here but it got left out

File: app/src/auth/components/SignInMethod.jsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    ${SignInMethodTitle} + ${SignInMethodIcon} {
+        margin-left: 16px;
+    }
+`
+
+const SignInMethod = ({ theme, ...props }) => (
+    <SignInMethodButton
+        type="button"
+        theme={theme || NormalTheme}
+        {...props}
```
[the remark this answers is not in the document]
Comment (juhah) [reply app/src/auth/components/SignInMethod.jsx:70]: Button types `reset` or `submit` don't really make sense here so I will leave that hardcoded, default theme is better declared in the props 👍

File: app/src/auth/components/Logo.jsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    &,
+    &:--enter,
+    &:--idle {
+        outline: 0;
+    }
+`
+
+const UnstyledLogo = ({ className, ...props }) => (
+    <div {...props}>
+        <StyledLink to={routes.root()} data-logo>
```
[the remark this answers is not in the document]
Comment (juhah) [reply app/src/auth/components/Logo.jsx:27]: It was used in the cypress tests, now the test is gone and so can this 👍

File: app/src/auth/components/LoginPage.jsx [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+    id: WALLET_CONNECT,
+    image: walletConnectLogo,
+    image2x: walletConnectLogo2x,
+    enabled: true,
+}]
+
+const LoginPage = () => {
+    const dispatch = useDispatch()
+    const [{ method, connecting, error }, setState] = useReducer((state, action) => (
+        (typeof handlers[action.type] === 'function') ? handlers[action.type](state, action) : state
```
[the remark this answers is not in the document]
Comment (juhah) [reply app/src/auth/components/LoginPage.jsx:110]: Maybe, maybe not... this part is uglier but I like being able to just return the state in the handler

**Pull request merged**